dropwizard-jersey/src/test/java/io/dropwizard/jersey/jackson/JsonProcessingExceptionMapperTest.java

```diff
@@ -40,10 +38,10 @@ protected void configureClient(ClientConfig config) {
     }
 
     @Test
-    public void returnsA500ForNonDeserializableRepresentationClasses() throws Exception {
+    public void returnsA400ForNonDeserializableRepresentationClasses() throws Exception {
         Response response = target("/json/broken").request(MediaType.APPLICATION_JSON)
                 .post(Entity.entity(new BrokenRepresentation(ImmutableList.of("whee")), MediaType.APPLICATION_JSON));
-        assertThat(response.getStatus()).isEqualTo(500);
+        isA400ForNonDeserializableRequest(response);
```

Comment:
I am not 100% happy with the `catch (Throwable cause)` approach, but I thought it would be better to come up with a patch to start the discussion.

So I understand your point: a BrokenRepresentation should be a 500. On the other hand, I want my bogus URLs (see #1539) to be treated as 400. Any suggestions on how to differentiate them?